### PR TITLE
[MM-18470] Add healthcheck api endpoint

### PR DESCRIPTION
**Pull request by hanzei**
This PR adds a `/health` API endpoint. It strictly follows https://tools.ietf.org/id/draft-inadarei-api-health-check-01.html.

I still have to add a test for this, but early feedback is appreciated.

Fixes: https://mattermost.atlassian.net/browse/MM-18470

### CODE REVIEW

File: internal/api/health_check.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+	"github.com/gorilla/mux"
+)
+
+var buildTag = ""
+var buildHash = ""
+
+type healthCheckResponse struct {
+	Status      string
+	Version     string
+	ReleaseID   string
```
Comment (hanzei) [inline internal/api/health_check.go]: This is the last git tag. Empty for now

File: internal/api/health_check.go [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+	Notes       []string
```
Comment (hanzei) [inline internal/api/health_check.go]: This contains the sha1 of the last commit. I didn't found a more elegant solution to add this information to the response.

File: internal/api/health_check.go [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+	Description string
+}
+
+func initHealthCheck(apiRouter *mux.Router, context *Context) {
+	addContext := func(handler contextHandlerFunc) *contextHandler {
```
Comment (hanzei) [inline internal/api/health_check.go]: I dislike that I had to re-add this here. Maybe it's worth to refactor into it's own method.

File: internal/api/plugins.go [hunk 4]
```diff
@@ -59,5 +59,6 @@ func handleGetPlugins(c *Context, w http.ResponseWriter, r *http.Request) {
 		plugins = []*model.Plugin{}
 	}
 
+	w.Header().Set("Content-Type", "application/json")
```
Comment (hanzei) [inline internal/api/plugins.go]: This fixes the `Content-Type` for the get plugins endpoint.

**Comment by hanzei** [comment]
@lieut-data I've added a test. PTAL

File: internal/api/health_check.go (hunk 2 above)
Comment (ali-farooq0) [reply internal/api/health_check.go]: spoke about it offline, suggested to use `Details` instead of `Notes`

**Comment by hanzei** [comment]
@ali-farooq0 I've made the changes. Please take a second look.

**Pull request merged**